src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -451,6 +451,22 @@ R"(#include <stdio.h>
                 }
             }
 
+            for (auto &item : x.m_symtab->get_scope()) {
+                ASR::symbol_t* var_sym = item.second;
+                if( ASR::is_a<ASR::ExternalSymbol_t>(*var_sym) ) {
+                    ASR::ExternalSymbol_t* v_ext = ASR::down_cast<ASR::ExternalSymbol_t>(var_sym);
+                    ASR::symbol_t* v_sym = v_ext->m_external;
+                    if (ASR::is_a<ASR::Variable_t>(*v_sym)) {
+                        ASR::Variable_t* v = ASR::down_cast<ASR::Variable_t>(v_sym);
+                        if( v->m_symbolic_value ) {
+                            this->visit_expr(*v->m_symbolic_value);
+                            decl += indent + "#define " + std::string(v_ext->m_name) + " " + src + "\n";
+                            src.clear();
```

Comment:
I think we have to be careful here.

What happens if `v_ext->m_name` is equal to "x" and then we have another variable like "expr`, will the `x` in it be substituted? I think it will be, so we do not want to do that. Also we do not want to have any such macro defines in header files. We currently do not generate them, but eventually we will.

I would not use C macros.

Reply:
> I would not use C macros.

So constant C variables should work for this case?

Reply:
Either that, or we can initialize the global variables when the program starts. I think both seems better than macros.